compiler/semstmts.nim

```diff
@@ -292,6 +292,9 @@ proc fitRemoveHiddenConv(c: PContext, typ: PType, n: PNode): PNode =
       result = newFloatNode(nkFloatLit, BiggestFloat r1.intVal)
       result.info = n.info
       result.typ = typ
+      if not floatRangeCheck(result.floatVal, typ):
+        localError(c.config, n.info, "cannot convert " & $result.floatVal &
```

Comment:
Please avoid the little code duplication by using a little helper proc.

Reply:
The duplicated error message? Or the whole check?

Reply:
Both solutions would be fine, use what you like better.